File: SyncKit/Classes/Realm/QSCloudKitSynchronizer+MultiRealmResultsController.m
//
//  QSCloudKitSynchronizer+MultiRealmResultsController.m
//  Pods
//
//  Created by Manuel Entrena on 28/05/2018.
//

#import "QSCloudKitSynchronizer+MultiRealmResultsController.h"
#import <SyncKit/SyncKit-Swift.h>

@implementation QSCloudKitSynchronizer (MultiRealmResultsController)

- (QSMultiRealmResultsController *)multiRealmResultsControllerWithClass:(Class)objectClass predicate:(NSPredicate *)predicate
{
    QSDefaultRealmProvider *provider = (QSDefaultRealmProvider *)self.adapterProvider;
    if (![provider isKindOfClass:[QSDefaultRealmProvider class]]) {
        return nil;
    }
    
    return [[QSMultiRealmResultsController alloc] initWithRealmProvider:provider fetchObjectClass:objectClass predicate:predicate];
}

@end

File: SyncKit/Classes/RealmSwift/QSCloudKitSynchronizer+MultiRealmResultsController.swift
//
//  QSCloudKitSynchronizer+MultiRealmResultsController.swift
//  Pods
//
//  Created by Manuel Entrena on 10/06/2018.
//

import Foundation
import RealmSwift

extension CloudKitSynchronizer {
    
    public func multiRealmResultsController<T: Object>(predicate: NSPredicate? = nil) -> MultiRealmResultsController<T>? {
        
        if let provider = self.adapterProvider as? DefaultRealmProvider {
            return MultiRealmResultsController(provider: provider, predicate: predicate)
        }
        return nil
    }
    
    public var realms: [Realm] {
        let provider = adapterProvider as? DefaultRealmProvider
        return provider?.realms.values.compactMap { try? Realm(configuration: $0) } ?? []
    }
}
